src/Aspire.Hosting.Azure.KeyVault/AzureKeyVaultResource.cs

```diff
@@ -13,21 +14,51 @@ namespace Aspire.Hosting.Azure;
 /// <param name="name">The name of the resource.</param>
 /// <param name="configureInfrastructure">Callback to configure the Azure resources.</param>
 public class AzureKeyVaultResource(string name, Action<AzureResourceInfrastructure> configureInfrastructure)
-    : AzureProvisioningResource(name, configureInfrastructure), IResourceWithConnectionString
+    : AzureProvisioningResource(name, configureInfrastructure), IResourceWithConnectionString, IKeyVaultResource
 {
     /// <summary>
     /// Gets the "vaultUri" output reference for the Azure Key Vault resource.
     /// </summary>
     public BicepOutputReference VaultUri => new("vaultUri", this);
 
-    private BicepOutputReference NameOutputReference => new("name", this);
+    /// <summary>
+    /// Gets the "name" output reference for the Azure Key Vault resource.
+    /// </summary>
+    public BicepOutputReference NameOutputReference => new("name", this);
+
+    /// <summary>
+    /// Gets the resource identifier of the Azure Key Vault resource.
+    /// </summary>
+    public BicepOutputReference Id => new("id", this);
 
     /// <summary>
     /// Gets the connection string template for the manifest for the Azure Key Vault resource.
     /// </summary>
     public ReferenceExpression ConnectionStringExpression =>
         ReferenceExpression.Create($"{VaultUri}");
 
+    /// <summary>
+    /// The secrets for the Azure Key Vault resource. Used in run mode to resolve
+    /// </summary>
+    public Dictionary<string, string> Secrets { get; } = [];
+
+    IDictionary<string, string> IKeyVaultResource.Secrets => Secrets;
```

Comment:
Should this just be `Dictionary<string, string>` so we don't need to hide it?

Reply:
Yes, it's really only for testing as well. I might make this internal.